### PR TITLE
Update midterm.md

### PR DESCRIPTION
**Pull request by silentbob78**
Frank's midterm

### CODE REVIEW

File: quizzes/midterm.md [hunk 1]
```diff
@@ -15,6 +15,14 @@
 	monkey = Monkey.new
```
Comment (seanculver) [inline quizzes/midterm.md:15]: 6/10 -  3 out of 5 correct.

File: quizzes/midterm.md [hunk 2]
```diff
@@ -15,6 +15,14 @@
...
+	
+	```
+	-You can't call Monkey to itself for inheriting
+	-my_vars is not called as a specific type of variable
+	-the first class needs to be called animal then there needs to be another class called Monkey that inherits
+	from the animal class
+	-the private method does not car the variable
+	```
 
 1. What is returned from the statement below?
```
Comment (seanculver) [inline quizzes/midterm.md:27]: 0/10

File: quizzes/midterm.md [hunk 3]
```diff
@@ -23,24 +31,31 @@
 
 	thoughts[2]
 	```
-
+```
+"Oh"
+```
 1. What is the result of this `fetch`?
```
Comment (seanculver) [inline quizzes/midterm.md:37]: 10/10

File: quizzes/midterm.md [hunk 4]
```diff
@@ -23,24 +31,31 @@
...
 
 	```ruby
 	my_hash = { my_key: 'value', another_key: 'another value' }
 	my_hash.fetch(:not_a_key_yet, 'default value')  # => ??????
 	```
+```
+'default value'
+```
 
 1. Give an example of a good variable name, and a bad variable name.
```
Comment (seanculver) [inline quizzes/midterm.md:47]: 10/10

File: quizzes/midterm.md [hunk 5]
```diff
@@ -23,24 +31,31 @@
...
-
+```
+Good: needed_variable  Bad: 3_needed_variables
+```
 
 
 
 
 1. What is the difference between `1..10` and `1...10`?
```
Comment (seanculver) [inline quizzes/midterm.md:55]: 10/10

File: quizzes/midterm.md [hunk 6]
```diff
@@ -68,20 +89,37 @@
 
 
 1. Demonstrate using multiple logical operators (||, &&)
```
Comment (seanculver) [inline quizzes/midterm.md:91]: 10/10

File: quizzes/midterm.md [hunk 7]
```diff
@@ -68,20 +89,37 @@
...
+=> code will run
 
+if (true || true)
+#some coding
+end
+=> code will run
+
+```
 
 1. What is the result of calling sort?
```
Comment (seanculver) [inline quizzes/midterm.md:105]: 10/10

File: quizzes/midterm.md [hunk 8]
```diff
@@ -68,20 +89,37 @@
...
 	["one fish", "two fish", "red fish", "blue fish"].sort
 	```
-
+```
+["blue fish", "one fish", "red fish", "two fish"]
+```
 
 1. Write a hash that has the following keys :name, :super_power, :weakness
-
+```
```
Comment (seanculver) [inline quizzes/midterm.md:115]: 10/10

File: quizzes/midterm.md [hunk 9]
```diff
@@ -68,20 +89,37 @@
...
+my_hash = { name: "Green Lantern", super_power: "Manipulate light", weakness: "Yellow"  }
+```
 
 1. T/F You can set a default value for a Hash
```
Comment (seanculver) [inline quizzes/midterm.md:119]: 10/10

File: quizzes/midterm.md [hunk 10]
```diff
@@ -68,20 +89,37 @@
...
-
+```
+True
+```
 
 
 1. What is the resulting hash?
```
Comment (seanculver) [inline quizzes/midterm.md:125]: 0/10

File: quizzes/midterm.md [hunk 11]
```diff
@@ -90,23 +128,40 @@
 	my_hash = { "a" => 100, "b" => 200, "c" => 300 }
 	my_hash.select {|key,value| key < "c"}
 	```
+	
+	```
+	{ nil }
+	```
 
 1. Write code that iterates through the numbers array and out`puts` the `KEY` for `each` of the values in the array.
```
Comment (seanculver) [inline quizzes/midterm.md:136]: 0/10

File: quizzes/midterm.md [hunk 12]
```diff
@@ -90,23 +128,40 @@
...
 	numbers = ['one', 'two', 'three']
 	```
+	
+	```
+	def init
+	  puts "numbers"
+	end
+	```
 
 1. Write code that `maps` the fruits array to a new array where all the values are `reverse`d. Eg. "apples" = "selppa"
```
Comment (seanculver) [inline quizzes/midterm.md:148]: 10/10. How did you get the above question wrong but this right? They are almost identical. :confused: =)

File: quizzes/midterm.md [hunk 13]
```diff
@@ -90,23 +128,40 @@
...
 
 	```ruby
 	fruits = [ "apples", "bananas", "carrots", "beets" ]
 	```
+	```
+	fruits.map { |fruits| fruits.reverse } 
+	```
 
 1. Which attr_xxxxx would you use in this example? (see below)
```
Comment (seanculver) [inline quizzes/midterm.md:157]: 10/10

File: quizzes/midterm.md [hunk 14]
```diff
@@ -90,23 +128,40 @@
...
-
+```
+attr_accessor
+```
 1. What would change in the tame method? (see below)
```
Comment (seanculver) [inline quizzes/midterm.md:161]: 0/10

File: quizzes/midterm.md [hunk 15]
```diff
@@ -90,23 +128,40 @@
...
-
+```
+I would add one more end to close out the tame method
+```
 1. Now go tame some lions, and don't get eaten! Use the class to invoke the `tame` method on an object instance. (see below)
```
Comment (seanculver) [inline quizzes/midterm.md:165]: 7/10. This wouldn't actually run but you got most of the way. You need to pass in arguments when you are creating the object.

File: quizzes/midterm.md [hunk 16]
```diff
@@ -135,6 +193,9 @@
 	"Always rzmzmbzr you arz absolutzly uniquz, likz zvzryonz zlsz"
 	```
 
+```
```
Comment (seanculver) [inline quizzes/midterm.md:196]: 10/10

File: quizzes/midterm.md [hunk 17]
```diff
@@ -142,13 +203,32 @@
 	  other_hash = { new_key: "new value", new_another: "I will overwrite or will I?"}
 	  my_hash.merge(other_hash)
 	```
-
+```
```
Comment (seanculver) [inline quizzes/midterm.md:206]: 10/10

File: quizzes/midterm.md [hunk 18]
```diff
@@ -142,13 +203,32 @@
...
+{ my_key=> "value", another_key=>: "another value", new_key=> "new value", new_another=> "I will overwrite or will I?" }
+```
 
 1. What is the first thing you should do with `Legacy Code?`. And no, you can't throw it out.
```
Comment (seanculver) [inline quizzes/midterm.md:210]: 0/10 - Write tests for it.

File: quizzes/midterm.md [hunk 19]
```diff
@@ -142,13 +203,32 @@
...
+```
+Isolate parts of the code to see what parts of it are depreciated
+```
 
 1. Who is the creator of Ruby?
```
Comment (seanculver) [inline quizzes/midterm.md:215]: 10/10

File: quizzes/midterm.md [hunk 20]
```diff
@@ -142,13 +203,32 @@
...
+```
+"Matz"
+```
 
 1. Write a class that uses a Global, Instance, Class, and local variable. Don't worry about what it does, just properly use the variables.
+```
+class Food
```
Comment (seanculver) [inline quizzes/midterm.md:222]: 9/10 - _variable not a good variable name

File: quizzes/midterm.md [hunk 21]
```diff
@@ -163,8 +243,22 @@
 	  end
 	end
 	```
+
```
Comment (seanculver) [inline quizzes/midterm.md:246]: 4/10 You got part of it. You needed to write the eat method.

File: quizzes/midterm.md [hunk 22]
```diff
@@ -163,8 +243,22 @@
...
+      	end
+      	
+      	bubbles = Monkey.new
+      	bubbles = Monkey.eat
+      	bubbles = Monkey.drink
+      	
+      	
+	```
 
 1. What is the difference between using `extend` and `include`?
```
Comment (seanculver) [inline quizzes/midterm.md:258]: 10/10

File: quizzes/midterm.md [hunk 23]
```diff
@@ -163,8 +243,22 @@
...
+```
+You would use include for instance methods and extend for class methods
+```
 
 1. Convert this into a case statement
```
Comment (seanculver) [inline quizzes/midterm.md:263]: 10/10

File: quizzes/midterm.md [hunk 24]
```diff
@@ -179,10 +273,31 @@
...
+	    puts 'eat'
+	  when "durian"
+	    puts 'no thanks!'
+	  else
+	    puts 'eat'
+	end
+		
+	```
+	
 1.  T  /  F Arrays can only hold `String` and `Hash` values?
```
Comment (seanculver) [inline quizzes/midterm.md:288]: 10/10

File: quizzes/midterm.md [hunk 25]
```diff
@@ -179,10 +273,31 @@
...
+```
+False
+```
 
 1.  What is the scope resolution operator?
```
Comment (seanculver) [inline quizzes/midterm.md:293]: 10/10

File: quizzes/midterm.md [hunk 26]
```diff
@@ -179,10 +273,31 @@
...
+```
+False
+```
 
 1.  What is the scope resolution operator? 
+```
+::
+```
 
 1. T / F Ruby expects that you might pass a hash as the last argument to a method.
```
Comment (seanculver) [inline quizzes/midterm.md:298]: 10/10

**Comment by seanculver** [comment]
Total score 206

File: quizzes/midterm.md (hunk 14 above)
Comment (silentbob78) [inline quizzes/midterm.md:161]: 5/10 for effort

File: quizzes/midterm.md (hunk 18 above)
Comment (silentbob78) [inline quizzes/midterm.md:210]: 10/10

**Pull request closed without merging**